awesomplete.js

```diff
@@ -253,20 +263,39 @@ _.SORT_BYLENGTH = function (a, b) {
 	return a < b? -1 : 1;
 };
 
-_.ITEM = function (text, input) {
-	var html = input === '' ? text : text.replace(RegExp($.regExpEscape(input.trim()), "gi"), "<mark>$&</mark>");
+_.ITEM = function (data, input) {
+	var html = input === '' ? data : data.replace(RegExp($.regExpEscape(input.trim()), "gi"), "<mark>$&</mark>");
 	return $.create("li", {
 		innerHTML: html,
 		"aria-selected": "false"
 	});
 };
 
-_.REPLACE = function (text) {
-	this.input.value = text;
+_.REPLACE = function (data) {
+	this.input.value = data.value;
+};
+
+/* eslint-disable no-unused-vars */
+_.DATA = function (data, input) {
+	return data;
 };
 
 // Private functions
 
+function convert(data) {
```

Comment:
We may also combine this with Suggestion shim, which I originally added for API compatibility only and to be removed in 2.0.

It might be OK if we have private method data conversion for Array/Object/String, but initially I thought it would be useful to have default data conversion for Array/Object/String cases with public API, so we can change them if necessary.